cloudinit/net/network_manager.py

```diff
@@ -370,8 +371,9 @@ def conn_filename(con_id, target=None):
 
 
 def available(target=None):
-    target_nm_dir = subp.target_path(target, NM_LIB_DIR)
-    return os.path.exists(target_nm_dir)
+    config_present = os.path.isfile(subp.target_path(target, path=NM_CFG_FILE))
+    nmcli_present = subp.which("nmcli", target=target)
```

Comment:
Wouldn't it also make sense to check if the NetworkManager service is enabled?

Reply:
Thanks for the comment @Conan-Kudo . I have updated the PR to take service status into account.

Reply:
We'd probably want this for Netplan too, right?

Reply:
Netplan doesn't have a service associated with it, so I don't think there's anything more to be done.